perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -316,6 +316,7 @@ private void deleteAnyGroup(PerunSession sess, Group group, boolean forceDelete)
 
 		List<Resource> assignedResources  = getPerunBl().getResourcesManagerBl()
 			.getResourceAssignments(sess, group, List.of()).stream()
+			.filter(res -> res.getSourceGroupId() == null) // only direct assignments
```

Comment:
There could be a problem when you want to delete a group and that group has an automatic assignment, which is not checked here. Same for delete resource.

Reply:
You are right. I am now removing all assignments and I created new method `removeAutomaticGroupFromResource` (in BL and impl layers) and made a few changes in existing method `removeGroupFromResource` in BL layer. Please, check these changes as well.